src/OS.c

```diff
@@ -6,104 +6,119 @@
  */
 
 #include "OS.h"
+
 #include "tm_stm32_gps.h"
 #include "GSM.h"
 #include "ESP01.h"
 #include "MPU6050.h"
 #include "MQTT_Tasks.h"
 #include "socket_Task.h"
+
+#include "MPU6050.h"
 #include "Accident.h"
+#include "task.h"
+
+#include "queue.h"
+#include "semphr.h"
+#include "stm32f4_discovery.h"
 /*create all tasks here*/
 
 
 extern TaskHandle_t handle_gsm_os_init;
 
-extern TaskHandle_t xTask_MPU6050_Read_RawData;
+TaskHandle_t xTask_MPU6050_Read_RawData;
+
+TaskHandle_t xTask_AccidentDetection_Declration;
+
+xQueueHandle Accident_Detection_Queue;
+xSemaphoreHandle Accident_Detection_semphr;
+
 
-//extern TaskHandle_t xTask_AccidentDetection_Declration;
 
 
 void OS_INIT(void){
 
+//STM_EVAL_LEDOn(LED3);
+	 Accident_Detection_Queue=xQueueCreate( 1, sizeof( unsigned int ) );
+	 vSemaphoreCreateBinary( Accident_Detection_semphr );
+
+	if((Accident_Detection_Queue!=NULL)&&(Accident_Detection_semphr!=NULL)){
+		xTaskCreate(GPS_update_task,
```

Comment:
no need to check semaphores here, this causes the whole system to be stalled (including the accident task itself), no need to prioritize the accident task using semaphores as it already have a higher priority than any other task

Reply:
okay nice hint finally